File: apps/uhd_app1.py
#!/usr/bin/env python2
# -*- coding: utf-8 -*-
#
# SPDX-License-Identifier: GPL-3.0
#
##################################################
# GNU Radio Python Flow Graph
# Title: Uhd App1
# Generated: Tue Sep 18 19:22:04 2018
# GNU Radio version: 3.7.12.0
##################################################

from gnuradio import blocks
from gnuradio import eng_notation
from gnuradio import gr
from gnuradio import uhd
from gnuradio.eng_option import eng_option
from gnuradio.filter import firdes
from optparse import OptionParser
import pmt
import time


class uhd_app1(gr.top_block):

    def __init__(self):
        gr.top_block.__init__(self, "Uhd App1")

        ##################################################
        # Variables
        ##################################################
        self.samp_rate = samp_rate = 32000

        ##################################################
        # Blocks
        ##################################################
        self.uhd_usrp_sink_0 = uhd.usrp_sink(
        	",".join(("", "")),
        	uhd.stream_args(
        		cpu_format="fc32",
        		channels=range(1),
        	),
        )
        self.uhd_usrp_sink_0.set_samp_rate(samp_rate)
        self.uhd_usrp_sink_0.set_center_freq(0, 0)
        self.uhd_usrp_sink_0.set_gain(0, 0)
        self.blocks_tagged_stream_to_pdu_0 = blocks.tagged_stream_to_pdu(blocks.complex_t, 'packet_len')
        self.blocks_file_source_1 = blocks.file_source(gr.sizeof_gr_complex*1, 'input', False)
        self.blocks_file_source_1.set_begin_tag(pmt.PMT_NIL)
        self.blocks_file_source_0 = blocks.file_source(gr.sizeof_gr_complex*1, '/dev/urandom', True)
        self.blocks_file_source_0.set_begin_tag(pmt.PMT_NIL)



        ##################################################
        # Connections
        ##################################################
        self.msg_connect((self.blocks_tagged_stream_to_pdu_0, 'pdus'), (self.uhd_usrp_sink_0, 'command'))
        self.connect((self.blocks_file_source_0, 0), (self.uhd_usrp_sink_0, 0))
        self.connect((self.blocks_file_source_1, 0), (self.blocks_tagged_stream_to_pdu_0, 0))

    def get_samp_rate(self):
        return self.samp_rate

    def set_samp_rate(self, samp_rate):
        self.samp_rate = samp_rate
        self.uhd_usrp_sink_0.set_samp_rate(self.samp_rate)


def main(top_block_cls=uhd_app1, options=None):

    tb = top_block_cls()
    tb.start()
    tb.wait()


if __name__ == '__main__':
    main()

File: examples/c++/uhd_app1.cc
// Include header files for each block used in flowgraph
#include <gnuradio/top_block.h>
#include <gnuradio/blocks/file_source.h>
#include <gnuradio/blocks/tagged_stream_to_pdu.h>
#include <gnuradio/uhd/usrp_sink.h>
#include <pmt/pmt.h>

//usage: uhd_app file_in
  
using namespace gr;

int main(int argc, char **argv)
{
  double samp_rate = 32000.;

  // top block
  top_block_sptr tb = make_top_block("uhd_app");
  
  // define blocks  
  static gr::uhd::usrp_sink::sptr uhd_usrp_sink_0 = gr::uhd::usrp_sink::make(::uhd::device_addr_t("192.168.1.40"),
    //::uhd::io_type_t{5},
    ::uhd::stream_args_t("", ""),
    ""
  );
  static blocks::tagged_stream_to_pdu::sptr blocks_tagged_stream_to_pdu_0 = blocks::tagged_stream_to_pdu::make(blocks::pdu::complex_t, "packet_len");
  static blocks::file_source::sptr blocks_file_source_1 = blocks::file_source::make(sizeof(gr_complex)*1, argv[1], false);
  static blocks::file_source::sptr blocks_file_source_0 = blocks::file_source::make(sizeof(gr_complex)*1, "/dev/urandom", true);
  
  // set some properties
  uhd_usrp_sink_0->set_samp_rate(samp_rate);
  uhd_usrp_sink_0->set_center_freq(0, 0);
  uhd_usrp_sink_0->set_gain(0, 0);
  blocks_file_source_0->set_begin_tag(pmt::PMT_NIL);
  blocks_file_source_1->set_begin_tag(pmt::PMT_NIL);        
  
  tb->connect(blocks_file_source_1, 0, blocks_tagged_stream_to_pdu_0, 0);
  tb->msg_connect(blocks_tagged_stream_to_pdu_0, "pdus", uhd_usrp_sink_0, "command");
  tb->connect(blocks_file_source_0, 0, uhd_usrp_sink_0, 0);

  tb->run();

  return 0;
}











